src/libcore/iter/iterator.rs

```diff
@@ -1745,6 +1745,38 @@ pub trait Iterator {
         }).break_value()
     }
 
+    /// Applies function to the elements of iterator and returns
+    /// the first non-none result.
+    ///
+    /// `iter.find_map(f)` is equivalent to `iter.filter_map(f).next()`.
+    ///
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(iterator_find_map)]
+    /// let a = ["lol", "NaN", "2", "5"];
+    ///
+    /// let mut first_number = a.iter().find_map(|s| s.parse().ok());
```

Comment:
The `mut` is not needed here, I believe.

Reply:
@meven seems plausible — submit a PR to fix it, please?

Reply:
PR opened at https://github.com/rust-lang/rust/pull/55389